lib/fetch/dataURL.js

```diff
@@ -258,9 +264,7 @@ function parseMIMEType (input) {
     /** @type {Map<string, string>} */
     parameters: new Map(),
     // https://mimesniff.spec.whatwg.org/#mime-type-essence
-    get essence () {
-      return `${this.type}/${this.subtype}`
-    }
+    essence: `${type}/${subtype}`
```

Comment:
This was the main driver for the improvements. Tried to find something in the spec that indicates this is a `getter` but sadly couldn't find any point indicating it. I might overlook it, but @KhafraDev can you confirm this approach is ok?

Reply:
This won't work as it can be overwritten...

Reply:
That was what I was missing then. Does that mean that the mime type object can be altered after creation, right?

Reply:
Yes, and it should be immutable.

Reply:
Actually there is more stuff here than should get getters?

Reply:
I might be wrong, but then it becomes a case of consistency, if you change subtype then essence need to be automatically updated, which is why a getter is necessary either way.

Reply:
>Actually there is more stuff here than should get getters?

You mean the PR itself, then yes? Otherwise, I might be missing something

>I might be wrong, but then it becomes a case of consistency, if you change subtype then essence need to be automatically updated, which is why a getter is necessary either way

Yeah, once you said that about that must be a getter, that is what I started to see, as most likely the issue is not on the getter itself but the recomputation of the `string` on every `get` call.

Reply:
It seems is actually more about the string computation, as if I replace it with the following class that computes the `essence` on instantiation, and on set/get (kinda caching), the results remains stable.

**Class**:
```js
class MimeType {
  #type = ''
  #subtype = ''
  #essence = ''
  #parameters = new Map()

  constructor ({ type, subtype }) {
    this.#type = type
    this.#subtype = subtype
    this.#essence = `${type}/${subtype}`
  }

  set type (value) {
    this.#type = value
    this.#essence = `${type}/${subtype}`
  }

  get type () {
    this.#type
  }

  set subtype (value) {
    this.#subtype = value
    this.#essence = `${type}/${subtype}`
  }

  get subtype () {
    return this.#subtype
  }

  get essence () {
    return this.#essence
  }

  get parameters () {
    return this.#parameters
  }
}```

**Scenario**
```
const str = 'application/json; charset=utf-8'

suite
  .add('util#MIMEType', function () {
    new util.MIMEType(str)
  })
  .add('undici#parseMIMEType', function () {
    parseMIMEType(str).essence
  })
  .add('undici#parseMIMEType(original)', function () {
    parseMIMETypeOriginal(str).essence
  })
  .add('fast-content-type-parse#parse', function () {
    fastContentType.parse(str)
  })
  .add('fast-content-type-parse#safeParse', function () {
    fastContentType.safeParse(str)
  })
  .on('cycle', function (event) {
    console.log(String(event.target))
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'))
  })
  .run({ async: true })```